src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -127,23 +153,23 @@ internal static bool IsEnabled()
                 if (activity != null)
                 {
                     s_diagnosticListener.StopActivity(activity, new ActivityStopData(
-                        responseTask?.Status == TaskStatus.RanToCompletion ? responseTask.Result : null,
+                        response,
                         // If request is failed or cancelled, there is no response, therefore no information about request;
                         // pass the request in the payload, so consumers can have it in Stop for failed/canceled requests
                         // and not retain all requests in Start
                         request,
-                        responseTask?.Status ?? TaskStatus.Faulted));
+                        taskStatus));
```

Comment:
This is a change in behavior.  Previously if the Task ended as canceled, this would be TaskStatus.Canceled... but the way you've written it now, it'll be TaskStatus.Faulted.

Reply:
Yes, it's a mistake. I slightly rewritten this part to get the status directly from `responseTask` in case of async (as it was before).